ocaml-backend/src/main/java/org/kframework/backend/ocaml/DefinitionToOcaml.java

```diff
@@ -286,18 +286,18 @@ public void initialize(CompiledDefinition def) {
                 throw KEMException.criticalError("Must have only one klabel with the \"thread\" attribute. Found: " + threadKLabels);
             }
             Rule matchThreadSet = Rule(IncompleteCellUtils.make(threadKLabels.iterator().next(), false, KVariable("Threads"), false), BooleanUtils.TRUE, BooleanUtils.TRUE);
-            this.matchThreadSet = convert(new Kompile(kompileOptions, globalOptions, files, kem).compileRule(def.kompiledDefinition, matchThreadSet));
+            this.matchThreadSet = convert(new Kompile(kompileOptions, kompileOptions.outerParsing, globalOptions, files, kem).compileRule(def.kompiledDefinition, matchThreadSet));
             Rule rewriteThreadSet = Rule(IncompleteCellUtils.make(threadKLabels.iterator().next(), false, KRewrite(KVariable("Threads"), KVariable("NewThreads")), false), BooleanUtils.TRUE, BooleanUtils.TRUE);
-            this.rewriteThreadSet = convert(new Kompile(kompileOptions, globalOptions, files, kem).compileRule(def.kompiledDefinition, rewriteThreadSet));
+            this.rewriteThreadSet = convert(new Kompile(kompileOptions, kompileOptions.outerParsing, globalOptions, files, kem).compileRule(def.kompiledDefinition, rewriteThreadSet));
             this.rewriteThreadSet = Rule(new TransformK() { public K apply(KVariable var) { return KVariable(var.name(), var.att().remove(Sort.class)); } }.apply(this.rewriteThreadSet.body()),
                     this.rewriteThreadSet.requires(), this.rewriteThreadSet.ensures());
         }
         KLabel stratCell = KLabel("<s>");
         if (mainModule.definedKLabels().contains(stratCell)) {
             Rule makeStuck = Rule(IncompleteCellUtils.make(stratCell, false, KRewrite(KSequence(), KApply(KLabel("#STUCK"))), true), BooleanUtils.TRUE, BooleanUtils.TRUE);
             Rule makeUnstuck = Rule(IncompleteCellUtils.make(stratCell, false, KRewrite(KApply(KLabel("#STUCK")), KSequence()), true), BooleanUtils.TRUE, BooleanUtils.TRUE);
-            this.makeStuck = convert(new Kompile(kompileOptions, globalOptions, files, kem).compileRule(def.kompiledDefinition, makeStuck));
-            this.makeUnstuck = convert(new Kompile(kompileOptions, globalOptions, files, kem).compileRule(def.kompiledDefinition, makeUnstuck));
+            this.makeStuck = convert(new Kompile(kompileOptions, kompileOptions.outerParsing, globalOptions, files, kem).compileRule(def.kompiledDefinition, makeStuck));
+            this.makeUnstuck = convert(new Kompile(kompileOptions, kompileOptions.outerParsing, globalOptions, files, kem).compileRule(def.kompiledDefinition, makeUnstuck));
```

Comment:
Probably these references should all use the injected outerParsing object instead of the reference from kompileOptions. That's going to be null.

Reply:
This doesn't actually matter because this method is always called by kompile, where it actually won't be null